File: frontend/ediary_cli/src/main/kotlin/org/srd/ediary_cli/model/Entry.kt
package org.srd.ediary_cli.model

import kotlinx.datetime.LocalDate
import kotlinx.serialization.Serializable

@Serializable
data class EntryInfoDTO(val id: Long, val title: String, val content: String, val createdDate: LocalDate)

@Serializable
data class EntryCreateDTO(val diaryID: Long, val title: String, val content: String)

@Serializable
data class EntryUpdateDTO(val title: String, val content: String)

File: frontend/ediary_vue/src/types/Entry.ts
export interface Entry{
    id: number;
    diaryID: number;
    title: string;
    content: string;
    createdDate: string;
}

export interface EntryInfoDTO {
    id: number;
    title: string;
    content: string;
    createdDate: string;
}

export interface EntryCreateDTO {
    diaryID: string;
    title: string;
    content: string;
}

export interface EntryUpdateDTO {
    title: string;
    content: string;
}

export interface EntryPermissionRes {
    allowed: boolean
}
